### PR TITLE
fix: compatibility with null post.updated

### PR DESCRIPTION
**Pull request by curbengh**
compatibility with https://github.com/hexojs/hexo/pull/3887 https://github.com/hexojs/hexo/pull/4278

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -340,6 +341,25 @@ describe('Feed generator', () => {
       feed_url: hexo.config.root + feedCfg.path
     }));
   });
+
+  it('no updated date', () => {
+    hexo.config.optional_updated = true;
```
Comment (curbengh) [inline test/index.js]: Most probably this config won't be applied as [`processor`](https://github.com/hexojs/hexo/blob/master/lib/plugins/processor/asset.js) doesn't run here.

File: test/index.js [hunk 2]
```diff
@@ -45,7 +45,8 @@ describe('Feed generator', () => {
     return Post.insert([
       {source: 'foo', slug: 'foo', content: '<h6>TestHTML</h6>', date: 1e8},
       {source: 'bar', slug: 'bar', date: 1e8 + 1},
-      {source: 'baz', slug: 'baz', title: 'With Image', image: 'test.png', date: 1e8 - 1}
+      {source: 'baz', slug: 'baz', title: 'With Image', image: 'test.png', date: 1e8 - 1},
+      {source: 'date', slug: 'date', title: 'date', date: 1e8 - 2, updated: undefined}
```
Comment (curbengh) [inline test/index.js]: will add a test case with valid `updated`.

**Comment by curbengh** [comment]
Pending release of Hexo 5.

File: package.json [hunk 3]
```diff
@@ -42,7 +42,7 @@
     "cheerio": "^0.22.0",
     "eslint": "^7.0.0",
     "eslint-config-hexo": "^4.1.0",
-    "hexo": "^4.0.0",
+    "hexo": "hexojs/hexo",
```
Comment (curbengh) [inline package.json:45]: Pending release of Hexo 5.

Comment (curbengh) [reply package.json:45]: @SukkaW okay with `"hexo": "hexojs/hexo"`, instead of `"^5.0.0"`?

Comment (SukkaW) [reply package.json:45]: @curbengh We can change this back to `^5.0.0` when release a newer version.

Comment (curbengh) [reply package.json:45]: I'll just wait then.

Comment (stevenjoezhang) [reply package.json:45]: dependabot PR: https://github.com/hexojs/hexo-generator-feed/pull/152

**Pull request merged**